Intersect.Server.Core/Entities/Projectile.cs

```diff
@@ -337,41 +364,39 @@ public bool CheckForCollision(ProjectileSpawn spawn)
                 var entities = mapInstance.GetEntities();
                 for (var z = 0; z < entities.Count; z++)
                 {
-                    if (entities[z] != null && entities[z] != spawn.Parent.Owner && entities[z].Z == spawn.Z &&
-                        (entities[z].X == Math.Round(spawn.X)) &&
-                        (entities[z].Y == Math.Round(spawn.Y)) &&
+                    var entity = entities[z];
+                    if (entity != null && entity != spawn.Parent.Owner && entity.Z == spawn.Z &&
+                        entity.X == roundedX && entity.Y == roundedY &&
                         (spawn.X != Owner.X || spawn.Y != Owner.Y))
                     {
-                        killSpawn = spawn.HitEntity(entities[z]);
+                        killSpawn = spawn.HitEntity(entity);
                         if (killSpawn && !spawn.ProjectileBase.PierceTarget)
                         {
                             return killSpawn;
                         }
                     }
-                    else
+                    else if (z == entities.Count - 1 && spawn.Distance >= Base.Range)
                     {
-                        if (z == entities.Count - 1)
-                        {
-                            if (spawn.Distance >= Base.Range)
-                            {
-                                killSpawn = true;
-                            }
-                        }
+                        killSpawn = true;
                     }
                 }
             }
 
-            return killSpawn;
+            return killSpawn || spawn?.Distance >= Base.Range;
         }
 
-        private float GetProjectileX(ProjectileSpawn spawn)
+        /// <summary>
+        /// Calculates the offset of a projectile based on the target's position and the spawn location.
+        /// </summary>
+        /// <param name="isXAxis">A boolean value indicating whether the calculation should be done horizontally (X-axis) or vertically (Y-axis).</param>
+        /// <param name="spawn">The projectile spawn information.</param>
+        /// <returns>The calculated offset value.</returns>
+        private float GetProjectileOffset(bool isXAxis, ProjectileSpawn spawn)
```

Comment:
too much nesting, from what I see only the first if can be inverted here

![image](https://github.com/AscensionGameDev/Intersect-Engine/assets/63019821/b3256f73-3fbf-4816-adda-cfb5196a0c7a)